src/AppliesHttpQuery.php

```diff
@@ -161,9 +162,29 @@ private function resolveColumn(Model $model, string $column): string
         return $resolved;
     }
 
+    /**
+     * Filter all columns which aren't in the searchableFields get parameter.
+     *
+     * @param  array  $columns
+     * @param  string  $searchableColumns
+     * @return array
+     */
+    private function filterColumns($columns, $searchFields)
+    {
+        $searchFields = explode(',', $searchFields);
+
+        // If no searchable fields are given we can presume to search in all
+        // columns specified in the model.
+        if (count($searchFields) === 0) {
+            return $columns;
+        }
+
+        return array_intersect($columns, $searchFields);
```

Comment:
I am afraid this won't be enough. This will not work in the following scenario:
```
// Comes from Model.
$columns = [
    'machines.name',
];

// Comes from request.
$search = [
    'machine.name',
];
```

This can happen to the following model:
```
{
    machine: {
        name: string
    }
}
```

The frontend will receive that model and does not know that `machine` should be `machines`.

This has already been fixed for ordering by making use of the `resolveColumn()`. This will turn `machine.name` to `machines.name`. Or even `machine.attribute.name` to `attributes.name`.

Reply:
But shouldn't the frontend developer provide the proper attribute?